Neos.ContentRepositoryRegistry/Classes/ContentRepositoryRegistry.php

```diff
@@ -248,6 +252,16 @@ private function buildUserIdProvider(ContentRepositoryId $contentRepositoryId, a
         return $userIdProviderFactory->build($contentRepositoryId, $contentRepositorySettings['userIdProvider']['options'] ?? []);
     }
 
+    private function buildPrivilegeProvider(ContentRepositoryId $contentRepositoryId, array $contentRepositorySettings, UserIdProviderInterface $userIdProvider, self $contentRepositoryRegistry): PrivilegeProviderInterface
+    {
+        isset($contentRepositorySettings['privilegeProvider']['factoryObjectName']) || throw InvalidConfigurationException::fromMessage('Content repository "%s" does not have privilegeProvider.factoryObjectName configured.', $contentRepositoryId->value);
+        $privilegeProviderFactory = $this->objectManager->get($contentRepositorySettings['privilegeProvider']['factoryObjectName']);
+        if (!$privilegeProviderFactory instanceof PrivilegeProviderFactoryInterface) {
+            throw InvalidConfigurationException::fromMessage('privilegeProvider.factoryObjectName for content repository "%s" is not an instance of %s but %s.', $contentRepositoryId->value, PrivilegeProviderFactoryInterface::class, get_debug_type($privilegeProviderFactory));
+        }
+        return $privilegeProviderFactory->build($contentRepositoryId, $contentRepositorySettings['userIdProvider']['options'] ?? [], $userIdProvider, $contentRepositoryRegistry);
```

Comment:
The `PrivilegeProvider` currently has a dependency to...
* the `UserIdProvider` to load policies of the authenticated user
* the `ContentRepositoryRegistry` to retrieve ContentStreams for the user (via `WorkspaceFinder`)..

This is nasty

Reply:
UserIdProvider is fine IMHO, ContentRepositoryRegistry should go away. What about making the $contentRepository available to the provider at runtime? would IMHO be easier and more consistent.

Reply:
@skurfuerst can you clarify what you mean by "available [...] at runtime"? A parameter of the `PrivilegeProvider::getPrivileges()` method? But than the ContentGraph would need to pass it

Reply:
yep, that's what I thought - similar to what we did in the command handlers. But you are right, the projection does not know it yet...